### PR TITLE
BCD table crashes shouldn't break the page

### LINKED ISSUE #2020: /en-US/docs/Web/API/CSSTransformComponent/toString crashes BCD table
The BCD table is supposed to have an error boundary around it to prevent this from happening. 

https://developer.allizom.org/en-US/docs/Web/API/CSSTransformComponent/toString

Let's fix the error boundary thing so that a breaking BCD table doesn't render the whole page broken. 
Then, let's fix the BCD bug.

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #2020

@ddbeck I can't request a review from you. I added you with `Write` permission to the repo but perhaps it's pending you invitation acceptance. Or perhaps some caching going on in the PR UI. Anyway...

@schalkneethling Can you please review the appears and can you also sanity test my PR. If you open `lazy-bcd-table.tsx` you'll see that there are 3 individual chances of NOT displaying the full BCD table but instead displaying an "error message". 
Can you please have a think about some better styling we can use. Perhaps there's something from minimalist or from the `ui` folder that I can use?

@ddbeck Can you sanity check the implementation and perhaps reflect on the "word smithing" of the error message it self. Here's what it looks like:
Screen Shot 2020-12-09 at 4 40 52 PM
It crashes hard [in stage](https://github.com/mdn/yari/issues/2020) because the sub-component shouldn't have been even attempted if the data isn't there. When open `node_modules/@mdn/browser-compat-data/api/CSSTransformComponent.json` I can clearly see that there's no entry for `toString`. I'm not judging that really. It's going to happen again sooner or later. And it can equally happen if you'd typed `{{Compat("api.CSSTransformComponent.toMatrixxxxtypo")}}` 

Whilst you guys can take a look at that I'm going to follow-up with a thing so that this is registered as a flaw. The author of that page should be made aware that `api.CSSTransformComponent.toString` doesn't exist in BCD.

### CODE REVIEW

**Comment by peterbe** [comment]
There. EIther you accepted the invitation or it was a matter of refreshing this page but now I'm able to request a review from your @ddbeck

**Comment by peterbe** [comment]
Ok, I fixed the problem with it not being flagged as a flaw. What happened was that it was seeking a property called `toString` which is not present in the bcd JSON files but it's truthy because it's a built-in prototype property. E.g.
```
> var some = {thing: 'sure'}
undefined
> if (!some.thing) { console.log('Some.thing is not there') }
undefined
> if (!some.definitelyNot) { console.log('Some.definitelyNot is not there') }
Some.definitelyNot is not there
undefined
> if (!some.toString) { console.log('Some.toString is not there') }
undefined
```
So I forced it to use `hasOwnProperty()` and that seems to solve it.

**Comment by peterbe** [comment]
Helps if I follow the `eslint` rules I added to the CI :) 
https://eslint.org/docs/rules/no-prototype-builtins

**Comment by peterbe** [comment]
Oh. I must have gotten something wrong. Now BCD tables aren't showing up at all :)
I'll figure it out tomorrow. I would still appreciate your review.

Note to self, see http://localhost:3000/en-US/docs/Web/CSS/transform-function/rotateY()#_flaws

File: client/src/document/lazy-bcd-table.tsx [hunk 1]
```diff
@@ -68,6 +74,15 @@ function LazyBrowserCompatibilityTableInner({ dataURL }: { dataURL: string }) {
   if (error) {
     return <p>Error loading BCD data</p>;
   }
+  if (!data.data) {
+    return (
+      <p>
+        Error loading BCD table because there's no known data for{" "}
+        <code>{query}</code>
+      </p>
```
Comment (ddbeck) [inline client/src/document/lazy-bcd-table.tsx:82]: Cribbing a bit from the current kumascript macro text. How about something like this?

```suggestion
      <p>
        No compatibility data found for <code>{query}</code>. Check the spelling or contribute data to <a href="https://github.com/mdn/browser-compat-data">mdn/browser-compat-data</a>.
      </p>
```

**Review by ddbeck — CHANGES_REQUESTED** [review]

**Comment by peterbe** [comment]
@ddbeck I accepted your edit. r?
@schalkneethling I'm really eager to merge this because it crashes the page. If you can't think of a quick and easy `className` to add or some SCSS I can copy-n-paste, can you please r+ and file a follow-up issue?

**Review by ddbeck — APPROVED** [review]
Thank you!

**Comment by peterbe** [comment]
@schalkneethling Since it turns out that the message is actually hidden if the BCD `query` doesn't resolve now. You actually don't even see the "error message" that Daniel and I debated. But there's an issue filed about reversing that so these error messages do become visible again. Would you mind filing a follow-up or something so we can include error messages that are appropriate for when things go wrong. It's also equally applicable for the KS macros that fail to render: https://github.com/mdn/yari/issues/1041 
So can you think up a good `className` I can use for display rendering problems?

**Comment by peterbe** [comment]
But for now, I'll merge this so we don't crash on those pages at least.

**Pull request merged**